pkg/containerutil/cp_linux.go

```diff
@@ -247,3 +275,46 @@ func mountSnapshotForContainer(ctx context.Context, client *containerd.Client, c
 	}
 	return tempDir, cleanup, nil
 }
+
+func getContainerMountInfo(ctx context.Context, con containerd.Container, containerPath string) (string, string, error) {
+	filePath := filepath.Clean(containerPath)
+	spec, err := con.Spec(ctx)
+	if err != nil {
+		return "", "", err
+	}
+	if spec.Root.Readonly {
+		return "", "", fmt.Errorf("container rootfs: %s is marked read-only", spec.Root.Path)
+	}
+
+	for _, mount := range spec.Mounts {
+		if isParentDir(filePath, mount.Destination) {
+			for _, option := range mount.Options {
+				if option == "ro" {
+					return "", "", fmt.Errorf("mount point %s is marked read-only", filePath)
+				}
+			}
+			return mount.Source, mount.Destination, nil
+		}
+	}
+	return "", "", nil
+}
+
+func isParentDir(filePath, potentialParent string) bool {
```

Comment:
```suggestion
	if filePath == "/"  ||  filePath == "" ||  potentialParent == "" {
		return false
	}
	filePath = filepath.Clean(filePath)
```
we need a base case here;  
`if filePath == "/"` : filepath is root, it does not have parent
`filePath = "" ||  potentialParent = ""`  this func can be used as utility for other context, so this check is need